lib/http/2/stream.rb

```diff
@@ -98,14 +99,14 @@ def receive(frame)
       case frame[:type]
       when :data
         # TODO: when receiving DATA, keep track of local_window.
+        @local_window -= frame[:payload].size
         emit(:data, frame[:payload]) if !frame[:ignore]
       when :headers, :push_promise
         emit(:headers, frame[:payload]) if !frame[:ignore]
       when :priority
         process_priority(frame)
       when :window_update
-        @remote_window += frame[:increment]
-        send_data
+        process_window_update(frame) if !frame[:ignore]
```

Comment:
Is the separate method call necessary? Seems like we're only calling this method from just this one place? Also the if check is duplicated.